File: Proyecto hacking/DESARROLLO/Generador_Datos/Data_Generator.java
package Generador_Datos;

import java.util.Random;
import java.util.Scanner;

public class Data_Generator {
    private static final String[] ALPHABET = {
        "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M",
        "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"
    };
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String response;
        
        do {
            System.out.println("¿Cuántas personas deseas generar? (Escribe 'salir' para terminar)");
            response = scanner.nextLine();
            
            if (response.equalsIgnoreCase("salir")) {
                break;
            }
            
            try {
                int numPeople = Integer.parseInt(response);
                
                for (int i = 0; i < numPeople; i++) {
                    System.out.println("Datos de la persona #" + (i + 1));
                    System.out.println("-----------------------------------------------------");
                    System.out.println("DNI: " + generateDNI());
                    System.out.println("Nombre: " + generateName());
                    System.out.println("Email: " + generateEmail());
                    System.out.println("Cuenta bancaria: " + generateBankAccount());
                    System.out.println("Fecha de nacimiento: " + generateDateOfBirth());
                    System.out.println("Contraseña: " + generatePassword());
                    System.out.println("-----------------------------------------------------");
                }
            } catch (NumberFormatException e) {
                System.out.println("Error: Ingresa un número válido o escribe 'salir' para terminar.");
            }
        } while (true);
        
        scanner.close();
    }
    
    public static String generateDNI() {
        StringBuilder dni = new StringBuilder();
        int length = 8; // Longitud del DNI (solo números)
        Random random = new Random();
        
        // Generar los dígitos del número de DNI
        for (int i = 0; i < length; i++) {
            dni.append(random.nextInt(10));
        }
        
        // Agregar una letra mayúscula al final
        dni.append(ALPHABET[random.nextInt(ALPHABET.length)]);
        
        return dni.toString();
    }
    
    public static String generateName() {
        Random random = new Random();
        String firstName = generateRandomName(FIRST_NAMES);
        String lastName = generateRandomName(LAST_NAMES);
        
        return firstName + " " + lastName;
    }
    
    public static String generateEmail() {
        Random random = new Random();
        String firstName = generateRandomName(FIRST_NAMES).toLowerCase();
        String lastName = generateRandomName(LAST_NAMES).toLowerCase();
        String domain = generateRandomName(EMAIL_DOMAINS);
        
        // Generar variaciones del correo electrónico con puntos o guiones bajos
        int variation = random.nextInt(3);
        String email;
        
        switch (variation) {
            case 0:
                email = firstName + "." + lastName + "@" + domain;
                break;
            case 1:
                email = firstName + "_" + lastName + "@" + domain;
                break;
            default:
                email = firstName + lastName + "@" + domain;
                break;
        }
        
        return email;
    }
    
    public static String generateBankAccount() {
        Random random = new Random();
        String bank = generateRandomName(BANKS);
        StringBuilder accountNumber = new StringBuilder();
        
        for (int i = 0; i < 10; i++) {
            accountNumber.append(random.nextInt(10));
            }
            return bank + " - " + accountNumber.toString();
    }

    public static String generateDateOfBirth() {
        Random random = new Random();
        int year = random.nextInt(50) + 1970; // Año aleatorio entre 1970 y 2019
        int month = random.nextInt(12) + 1; // Mes aleatorio entre 1 y 12
        int day = random.nextInt(28) + 1; // Día aleatorio entre 1 y 28
        return String.format("%04d-%02d-%02d", year, month, day);
    }

    public static String generatePassword() {
        StringBuilder password = new StringBuilder();
        int length = 8; // Longitud de la contraseña
        Random random = new Random();
        String characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()";

        for (int i = 0; i < length; i++) {
            password.append(characters.charAt(random.nextInt(characters.length())));
        }

        return password.toString();
    }

    private static String generateRandomName(String[] names) {
        Random random = new Random();
        return names[random.nextInt(names.length)];
    }

    private static final String[] FIRST_NAMES = {
        "Juan", "María", "José", "Laura", "Carlos", "Sofía", "Luis", "Ana", "Pedro", "Marta",
        "Alejandro", "Isabella", "Diego", "Valentina", "Gabriel"
    };
    
    private static final String[] LAST_NAMES = {
        "Gómez", "Rodríguez", "Fernández", "López", "Pérez", "González", "Martínez", "Sánchez", "Romero", "Torres",
        "Ortega", "Hernández", "Silva", "Ramírez", "Chavez"
    };
    
    private static final String[] EMAIL_DOMAINS = {
        "gmail.com", "yahoo.com", "hotmail.com", "outlook.com", "aol.com",
        "icloud.com", "mail.com", "live.com", "protonmail.com", "example.com"
    };
    
    private static final String[] BANKS = {
        "Banco A", "Banco B", "Banco C", "Banco D", "Banco E",
        "Banco F", "Banco G", "Banco H", "Banco I", "Banco J"
    };
    
}

File: Python/Data_generator.py
import random
from colorama import Fore, Style, init
from Python.ui import draw_banner, pause

init(autoreset=True)
COMMON_ASCII = r"""
 Data Gen
"""
BANNER_COLOR = "\033[92m"

ALPHABET = list("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
FIRST_NAMES = [
    "Juan", "Maria", "Jose", "Laura", "Carlos", "Sofia", "Luis", "Ana",
    "Pedro", "Marta", "Alejandro", "Isabella", "Diego", "Valentina",
    "Gabriel", "Fernando", "Adriana", "Miguel", "Elena", "Ricardo",
    "Beatriz", "Javier", "Claudia", "Raul", "Natalia", "Roberto",
    "Monica", "Daniel", "Carolina", "Hugo", "Victoria", "Fabiola",
    "Eduardo", "Carmen", "Angel", "Renata", "Pablo", "Lucia",
    "Olivia", "Andres", "Silvia", "Francisco",
]
LAST_NAMES = [
    "Gomez", "Rodriguez", "Fernandez", "Lopez", "Perez", "Gonzalez",
    "Martinez", "Sanchez", "Romero", "Torres", "Ortega", "Hernandez",
    "Silva", "Ramirez", "Chavez", "Luna", "Mendoza", "Guerrero", "Cruz",
    "Vargas", "Cabrera", "Cortez", "Roman", "Castaneda", "Zapata",
    "Aguirre", "Moreno", "Delgado", "Ramos", "Orozco", "Soto", "Molina",
    "Cisneros", "Fuentes", "Vega", "Gimenez", "Rosales", "Flores",
    "Valdez", "Acosta", "Herrera", "Nunez", "Carrillo", "Lara", "Escobar",
]
BANKS = [
    "Banco Santander", "BBVA", "CaixaBank", "Banco Sabadell", "Bankinter",
    "Kutxabank", "Abanca", "Unicaja Banco", "Ibercaja", "Deutsche Bank",
    "Barclays", "HSBC", "BNP Paribas", "Citibank",
]
EMAIL_DOMAINS = [
    "gmail.com", "outlook.com", "hotmail.com", "protonmail.com", "example.com",
]
def banner():
    draw_banner("", COMMON_ASCII, "Datos ficticios para pruebas", BANNER_COLOR)


def generate_dni():
    base = "".join(str(random.randint(0, 9)) for _ in range(8))
    return f"{base}{random.choice(ALPHABET)}"


def generate_name_and_email():
    first_name = random.choice(FIRST_NAMES)
    last_name = random.choice(LAST_NAMES)
    domain = random.choice(EMAIL_DOMAINS)
    alias = f"{first_name.lower()}.{last_name.lower()}{random.randint(10, 99)}"
    return f"{first_name} {last_name}", f"{alias}@{domain}"


def generate_bank_account():
    iban_body = "".join(str(random.randint(0, 9)) for _ in range(20))
    bank = random.choice(BANKS)
    return f"{bank} - ES{random.randint(10, 99)} {iban_body[:4]} {iban_body[4:8]} {iban_body[8:12]} {iban_body[12:16]} {iban_body[16:]}"


def generate_date_of_birth():
    year = random.randint(1970, 2005)
    month = random.randint(1, 12)
    day = random.randint(1, 28)
    return f"{year:04d}-{month:02d}-{day:02d}"


def generate_password():
    chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*"
    return "".join(random.choice(chars) for _ in range(12))


def data_generator_main():
    while True:
        banner()
        response = input("Cuantas identidades ficticias deseas generar? ('n' para volver): ").strip()
        if response.lower() == "n":
            return
        try:
            num_people = int(response)
            if num_people <= 0:
                raise ValueError
        except ValueError:
            pause(Fore.RED + "Introduce un numero entero positivo. Pulsa Enter para continuar...")
            continue

        banner()
        for i in range(1, num_people + 1):
            name, email = generate_name_and_email()
            print(Fore.CYAN + f"Persona #{i}")
            print("-" * 60)
            print("DNI:", generate_dni())
            print("Nombre:", name)
            print("Email:", email)
            print("Cuenta bancaria:", generate_bank_account())
            print("Fecha de nacimiento:", generate_date_of_birth())
            print("Password:", generate_password())
            print("-" * 60)
        pause()


if __name__ == "__main__":
    data_generator_main()
